### PR TITLE
Opening KCL sample now overwrites units

### LINKED ISSUE #3958: Open Sample Errors on Car Wheel Sample
### Description
Using "Open Sample" on samples that use `in()` or `mm()` (or equivalent) causes errors. When I opened the Car Wheel sample, I got an error becuase the units were in mm(), but it is expecting inches. We need to also bring in the units along with the sample. 

### Version
v0.25.4

### Recordings/Screenshots
Screenshot 2024-09-23 at 1 52 36 PM

### PR DESCRIPTION
**Pull request by franknoirot**
Closes #3958 by implementing basic settings overwrite behavior when importing samples and updating the review messages to warn users of this behavior.

@jessfraz mentioned it would be good to have a "Create new project" option when opening a KCL sample so we have a non-destructive user path. I agree, but there is a little refactor needed to enable that so I split it out into #3969 as an enhancement.

### CODE REVIEW

File: e2e/playwright/testing-samples-loading.spec.ts [hunk 1]
```diff
@@ -15,8 +16,8 @@ test.afterEach(async ({ page }, testInfo) => {
 
 test.describe('Testing in-app sample loading', () => {
   /**
-   * Note this test implicitly depends on the KCL sample "flange-with-patterns.kcl"
-   * and its title. https://github.com/KittyCAD/kcl-samples/blob/main/flange-with-patterns/flange-with-patterns.kcl
+   * Note this test implicitly depends on the KCL sample "car-wheel.kcl",
```
Comment (lf94) [inline e2e/playwright/testing-samples-loading.spec.ts:19]: Don't we mean explicitly here

File: src/components/FileMachineProvider.tsx [hunk 2]
```diff
@@ -306,6 +308,18 @@ export const FileMachineProvider = ({
...
+          // with the sample's setting.
+          if (data.sampleUnits) {
+            settings.send({
+              type: 'set.modeling.defaultUnit',
+              data: {
+                level: 'project',
+                value: data.sampleUnits,
+              },
+            })
+          }
```
Comment (lf94) [inline src/components/FileMachineProvider.tsx:322]: What happens when a user goes back to a file they own outside of the sample? Will the units revert?

File: src/lib/kclCommands.ts [hunk 3]
```diff
@@ -44,20 +52,45 @@ export function kclCommands(
...
+              )
+              let projectSettingsPayload: ReturnType<
+                typeof projectConfigurationToSettingsPayload
+              > = {}
+              if (!err(parsedProjectSettings)) {
+                projectSettingsPayload = projectConfigurationToSettingsPayload(
+                  parsedProjectSettings
+                )
+              }
+
```
Comment (lf94) [inline src/lib/kclCommands.ts:84]: Could we report something that there was an error? Easier to debug when something goes wrong

**Review by lf94 — COMMENTED** [review]
LGTM. Did not run the code yet though. These types of changes are always tricky when it comes to overriding settings. Please do not merge until someone or myself has tested this locally <3

**Review by jtran — APPROVED** [review]

**Comment by jgomez720** [comment]
Switching to the car wheel or wheel rotor did exactly as I expect, which was change the units to inches. Sweet.

Going back to a sample that uses mm switched it back to mm. Good to me

**Review by jgomez720 — APPROVED** [review]

**Comment by lf94** [comment]
@jgomez720 what about going back to a file you own

**Comment by jgomez720** [comment]
Just tested locally. Switching to another project switched the units to that project. It looks good to me

**Comment by lf94** [comment]
I mean within the same project.

1. Open a project with a file already
2. Open a sample 
3. Navigate back to the original file

Do the units switch back? Do we all agree it's not a problem if they don't?

**Comment by jgomez720** [comment]
No, it does not. 
- created a file, set units to "mm"
- open sample -> car-wheel -> units changed to "in"
- went back to the original file, units were still "in"

**Comment by jessfraz** [comment]
@lf94 in the future i proposed this since this is how the cli works https://github.com/KittyCAD/modeling-app/issues/3960

**Comment by jessfraz** [comment]
which would also fix this issue

**Comment by jessfraz** [comment]
AND allow us to not even have to tell the user about the units since we'd just throw it in a subdir with a project.toml with those units. anyways, that issue clears a lot of weird shit

**Comment by jessfraz** [comment]
and sets us up nicely for imports and assemblies

**Comment by lf94** [comment]
So we're good with the current behavior right

**Comment by lf94** [comment]
Are we going to merge

**Comment by Irev-Dev** [comment]
yea @franknoirot I think we're good to merge this one?

**Comment by franknoirot** [comment]
Yeah sounds good, we have the longer-term solution in an issue so I'm good to merge.

**Pull request merged**